script.xbmc.unpausejumpback/resources/lib/common.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+# Handy utility functions for Kodi Addons
+# By bossanova808
+# Free in all senses....
+# VERSION 0.1.1
+# (Matrix on)
+
+import xbmc
+import xbmcaddon
+import sys
+import traceback
+
+ADDON = xbmcaddon.Addon()
+ADDON_NAME = ADDON.getAddonInfo('name')
+ADDON_ID = ADDON.getAddonInfo('id')
+ADDON_ICON = ADDON.getAddonInfo('icon')
+ADDON_AUTHOR = ADDON.getAddonInfo('author')
+ADDON_VERSION = ADDON.getAddonInfo('version')
+ADDON_ARGUMENTS = str(sys.argv)
+CWD = ADDON.getAddonInfo('path')
+LANGUAGE = ADDON.getLocalizedString
+PROFILE = xbmc.translatePath(ADDON.getAddonInfo('profile'))
+KODI_VERSION = xbmc.getInfoLabel('System.BuildVersion')
+USER_AGENT = "Mozilla/5.0 (Windows; U; Windows NT 5.1; fr; rv:1.9.0.1) Gecko/2008070208 Firefox/3.6"
+
+
+def log(message, exception_instance=None, level=xbmc.LOGDEBUG):
+    """
+    Log a message to the Kodi debug log, if debug logging is turned on.
+
+    :param message: required, the message to log
+    :param exception_instance: optional, an instance of some Exception
+    :param level: optional, the Kodi log level to use, default LOGDEBUG
+    """
+
+    message = f'### {ADDON_NAME} {ADDON_VERSION} - {message}'
+    message_with_exception = message + f' ### Exception: {traceback.format_exc(exception_instance)}'
+
+    if exception_instance is None:
+        xbmc.log(message, level)
+    else:
+        xbmc.log(message_with_exception, level)
+
+
+def log_notice(message, exception_instance=None):
+    """
+    Log a message at the LOGNOTICE level, i.e. even if Kodi debugging is not turned on. Use sparingly.
+
+    :param message: required, the message to log
+    :param exception_instance: optional, an instance of some Exception
+    """
+
+    log(message, exception_instance, level=xbmc.LOGNOTICE)
```

Comment:
careful as lognotice was removed in matrix: https://codedocs.xyz/xbmc/xbmc/group__python__xbmc.html#ga3cd3eff2195727144ad6fbabd4744961